File: LeetCodeProblemsGo/LeetCode2145CountTheHiddenSequences/LeetCode2145CountTheHiddenSequences.go
package LeetCode2145CountTheHiddenSequences

func numberOfArrays(differences []int, lower int, upper int) int {
	max := 0
	min := 0
	sum := 0
	for _, difference := range differences {
		sum += difference
		if sum > max {
			max = sum
		}
		if sum < min {
			min = sum
		}
	}
	if (max - min) > (upper - lower) {
		return 0
	}
	return (upper - lower) - (max - min) + 1
}

File: LeetCodeProblems/Problems/LeetCode2145CountTheHiddenSequences.cs
namespace LeetCodeProblems.Problems;

public class LeetCode2145CountTheHiddenSequences
{
    public int NumberOfArrays(int[] differences, int lower, int upper)
    {
        long max = 0;
        long min = 0;
        long sum = 0;
        foreach (var difference in differences)
        {
            sum += difference;
            if (sum > max)
                max = sum;
            if (sum < min)
                min = sum;
        }

        if (max - min > upper - lower)
            return 0;
        int deltaMax =upper-(int)max;
        int deltaMin = lower-(int)min;
        return int.Abs(deltaMax - deltaMin)+1;
        
        
    }
}